modules/test/playwright/pages/commerce/commerce-order-content-web/pendingOrdersPage.ts

```diff
@@ -0,0 +1,97 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import {Locator, Page} from '@playwright/test';
+
+import {CommerceLayoutsPage} from '../commerceLayoutsPage';
+
+export class PendingOrdersPage {
+	readonly addPendingOrdersLabel: Locator;
+	readonly addWidgetButton: Locator;
+	readonly layoutsPage: CommerceLayoutsPage;
+	readonly orderItemActionsButton: Locator;
+	readonly  orderItemEditActionButton: Locator;
+	readonly page: Page;
+	readonly pageLabel: Locator;
+	readonly pageTitle: Locator;
+	readonly panelList: Locator;
+	readonly searchFormInput: Locator;
+	readonly selectPendingOrdersPageInput: Locator;
+	readonly viewButton: Locator;
+
+	constructor(page: Page) {
+
+		this.addPendingOrdersLabel = page
+			.getByTestId('addPanelTabItem')
+			.filter({hasText: /^Open Carts$/})
+			.getByRole('button', {exact: true, name: 'Add Content'});
+		this.addWidgetButton = page.getByTestId('add');
```

Comment:
Personally I don't like that the "creation of the page content" (so adding the widget) is inside the page that uses it. What do you think to have a method (I think in the CommerceLayoutPage) that giving the page and the widget name adds it to the page?

Reply:
I don't know I don't particularly like big fat unspecialised component, do we already have the pattern for this? Otherwise I think we should decide one in the next ET